### ISSUE TITLE
Application Image Specification

### ISSUE THREAD

**Issue by crudbug** [joyent/containerpilot#231]
I was not sure where to ask this. Is there a mailing list for containerpilot ?

I am doing my research on containers. 

We now have a standard container image format. Is it right to say containerpilot is trying standardize multi-node distributed application format with autopilot pattern that are self-aware of the environment they are running in ?  

RedHat is developing http://projectatomic.io , they have an application format called nulecule - http://www.projectatomic.io/docs/nulecule/ 

Can we compare nulecule specification with containerpilot config ?

**Comment by tgross** [comment]
> Is there a mailing list for containerpilot ?

There isn't one, but feel free to have these kinds of discussions here or on the appropriate github.com/autopilotpattern repos.

> We now have a standard container image format.

A bit of a side bar, but in what sense do you mean? As far as I can tell there are still several container image formats in use.

> Is it right to say containerpilot is trying standardize multi-node distributed application format with autopilot pattern that are self-aware of the environment they are running in ?

I'd veer away from using the term "standardize" myself. The Autopilot Pattern is an approach to design rather than a specification. All ContainerPilot provides is a convenient mechanism to orchestrate that design pattern. All the behaviors that ContainerPilot activates are owned by the end-user developer (as they should be!).

**Comment by crudbug** [comment]
Yes, there are multiple image formats. I was referring to https://www.opencontainers.org initiative. 

After thinking more about it, using "Pattern" is the right way to look at it. Sitting on top of container runtime, ContainerPilot provides an elegant solution to developers for creating distributed apps. 

It will be useful to integrate 12 factor application pattern - https://12factor.net , into autopilot pattern.

**Comment by tgross** [comment]
> It will be useful to integrate 12 factor application pattern - https://12factor.net , into autopilot pattern.

Stateless applications are honestly rather trivial from an orchestration standpoint. A lot of effort has been made in the container ecosystem to focus on stateless applications because they're an easy first step. We've done some stateless applications with the Autopilot Pattern but our focus from the very beginning was making sure we had a solution that worked just as well for applications with persistent state. That's why we did [mysql](https://github.com/autopilotpattern/mysql), [couchbase](https://github.com/autopilotpattern/couchbase), and [elasticsearch](https://github.com/autopilotpattern/elasticsearch)/[elk](https://github.com/autopilotpattern/elk).

Just out of curiosity and to make the discussion more concrete, what are you working on @crudbug?
